### PR TITLE
Feature multi page component api

### PR DESCRIPTION
**Pull request by a-szotyori**
Refactored `MultiPageReviewActivity` and created a `MultiPageReviewFragment` to be used with the Component API. Also moved classes and interfaces into their own files.

Updated the Component API example app to work with multi-page.

Introduced a memory store for the multi-page document instance for easy sharing across fragments called `ImageMultiPageDocumentMemoryStore`. It is owned by the `GiniVision` instance.

### How to test
Run the Component API example app.

### Ticket 
Issue #172

### CODE REVIEW

File: ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+    private PreviewsAdapter mPreviewsAdapter;
+    private PreviewsPageChangeHandler mPreviewsPageChangeHandler;
+    private TextView mPageIndicator;
+    private RecyclerView mThumbnailsRecycler;
+    private ThumbnailsAdapter mThumbnailsAdapter;
+    private RecyclerView.SmoothScroller mThumbnailsScroller;
+    private ThumbnailChangeListener mThumbnailChangeListener;
+    private ImageButton mButtonNext;
+    private ImageButton mRotateButton;
+    private ImageButton mReorderButton;
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java]: You don't need this button anymore 👍

File: ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+        mDeleteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                onDeleteButtonClicked();
+            }
+        });
+    }
+
+    private void onDeleteButtonClicked() {
+        final int deletedItem = mPreviewsPager.getCurrentItem();
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java]: This method does a lot of things. It deletes locally, deletes remotely and updates the thumbnails collection view.

File: componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java [hunk 3]
```diff
@@ -175,12 +174,12 @@ private void startGiniVisionCompat() {
                     @Override
                     public void permissionGranted() {
                         // NOTE: on Android 6.0 and later the camera permission is required before checking the requirements
-                        final RequirementsReport report = GiniVisionRequirements.checkRequirements(
-                                MainActivity.this);
-                        if (!report.isFulfilled()) {
-                            showUnfulfilledRequirementsToast(report);
-                            return;
-                        }
+//                        final RequirementsReport report = GiniVisionRequirements.checkRequirements(
```
Comment (kikettas) [inline componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java]: If you don't want it, delete it 🗑

File: componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java [hunk 4]
```diff
@@ -205,12 +204,12 @@ private void startGiniVisionStandard() {
...
```
Comment (kikettas) [inline componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java]: Same here!

File: ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java (hunk 1 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/review/multipage/MultiPageReviewFragment.java]: 👍 #193

File: componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java (hunk 3 above)
Comment (a-szotyori) [reply componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java]: Good catch. I commented it out to allow running on emulators. I will uncomment it and only show unfulfilled requirements without blocking execution of the GVL.

File: componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java (hunk 4 above)
Comment (a-szotyori) [reply componentapiexample/src/main/java/net/gini/android/vision/component/MainActivity.java]: Same as above and will also update it in the Screen API example app's `MainActivity`.

**Comment by a-szotyori** [comment]
Thanks for your review! I made some changes according to your suggestions. 👍

**Pull request merged**